code/game/objects/items/weapons/grenades/ghettobomb.dm

```diff
@@ -99,6 +142,42 @@
 				H.legcuffed = null
 	qdel(src)
 
+
+/obj/item/weapon/grenade/iedcasing/proc/process_shrapnel()
+
+	if(shrapnel_list.len > 0)
+		var/atom/target
+		var/atom/curloc = get_turf(src)
+		var/list/possible_targets= trange(7, curloc)
+		var/list/bodyparts = list("head","chest","groin","l_arm","r_arm","l_hand","r_hand","l_leg","r_leg","l_foot","r_foot")
+		for(var/obj/item/shrapnel in shrapnel_list)
+			if(istype(shrapnel, /obj/item/ammo_casing))// If the shrapnel is a bullet casing it will be fired
```

Comment:
The whole thing, and because that's how OO design works. Typechecking should be avoided.